### PR TITLE
try debug install step

### PR DESCRIPTION
**Pull request by davidhewitt**
Similar to #80, I want to see what pip manages to install on the broken platforms.

### CODE REVIEW

**Comment by davidhewitt** [comment]
Contents of the built wheel.

Seems to be missing the python files at this point, so it looks like a `maturin` issue. cc @messense

```
2023-12-22T22:53:53.9417945Z File Name                                             Modified             Size
2023-12-22T22:53:53.9421012Z pydantic_core-2.14.6.dist-info/METADATA        2023-12-22 22:53:52         6514
2023-12-22T22:53:53.9422504Z pydantic_core-2.14.6.dist-info/WHEEL           2023-12-22 22:53:52           99
2023-12-22T22:53:53.9491549Z pydantic_core-2.14.6.dist-info/license_files/LICENSE 2023-12-22 22:53:52         1080
2023-12-22T22:53:53.9493014Z pydantic_core/_pydantic_core.cpython-310-x86_64-linux-gnu.so 2023-12-22 22:53:52      4988336
2023-12-22T22:53:53.9493709Z pydantic_core-2.14.6.dist-info/RECORD          2023-12-22 22:53:52          456
```

File: recipe/meta.yaml [hunk 1]
```diff
@@ -13,7 +13,11 @@ build:
   script:
     # PyPy has weird sysconfigdata name
     - rm -f $PREFIX/lib/pypy$PY_VER/_sysconfigdata.py  # [build_platform != target_platform and target_platform == "linux-ppc64le"]
-    - {{ PYTHON }} -m pip install . -vv
+    - RUST_LOG=debug {{ PYTHON }} -m pip install . --no-build-isolation -vvv
+    - {{ PYTHON }} -m zipfile --list target/wheels/*.whl
+    - ls -l $PREFIX/lib/python$PY_VER/site-packages
+    - ls -l $PREFIX/lib/python$PY_VER/site-packages/pydantic_core
+    - {{ PYTHON }} -c 'import pydantic_core; print("\n\n===PYDANTIC CORE===\n\n", dir(pydantic_core))'
   number: 0
```
Comment (hmaarrfk) [inline recipe/meta.yaml:21]:
```suggestion
  number: 0
  skip: true  # [py!=310]
```

**Comment by hmaarrfk** [comment]
The thing is, that I've been trying to downgrade maturin locally, and I can still reproduce the issue.

**Comment by davidhewitt** [comment]
Looking at the maturin log, this looks like the cause:

```
2023-12-22T23:13:41.6790249Z   [2m2023-12-22T23:13:41.078672Z[0m [34mDEBUG[0m [2mignore::walk[0m[2m:[0m ignoring /home/conda/feedstock_root/build_artifacts/pydantic-core_1703286521520/work/python/pydantic_core/core_schema.py: Ignore(IgnoreMatch(Gitignore(Glob { from: Some("/home/conda/feedstock_root/.gitignore"), original: "*", actual: "**/*", is_whitelist: false, is_only_dir: false })))
2023-12-22T23:13:41.6791204Z   [2m2023-12-22T23:13:41.078697Z[0m [34mDEBUG[0m [2mignore::walk[0m[2m:[0m ignoring /home/conda/feedstock_root/build_artifacts/pydantic-core_1703286521520/work/python/pydantic_core/__pycache__: Ignore(IgnoreMatch(Gitignore(Glob { from: Some("/home/conda/feedstock_root/.gitignore"), original: "*", actual: "**/*", is_whitelist: false, is_only_dir: false })))
2023-12-22T23:13:41.6792170Z   [2m2023-12-22T23:13:41.078729Z[0m [34mDEBUG[0m [2mignore::walk[0m[2m:[0m ignoring /home/conda/feedstock_root/build_artifacts/pydantic-core_1703286521520/work/python/pydantic_core/_pydantic_core.pyi: Ignore(IgnoreMatch(Gitignore(Glob { from: Some("/home/conda/feedstock_root/.gitignore"), original: "*", actual: "**/*", is_whitelist: false, is_only_dir: false })))
2023-12-22T23:13:41.6793146Z   [2m2023-12-22T23:13:41.078738Z[0m [34mDEBUG[0m [2mignore::walk[0m[2m:[0m ignoring /home/conda/feedstock_root/build_artifacts/pydantic-core_1703286521520/work/python/pydantic_core/__init__.py: Ignore(IgnoreMatch(Gitignore(Glob { from: Some("/home/conda/feedstock_root/.gitignore"), original: "*", actual: "**/*", is_whitelist: false, is_only_dir: false })))
2023-12-22T23:13:41.6794107Z   [2m2023-12-22T23:13:41.078744Z[0m [34mDEBUG[0m [2mignore::walk[0m[2m:[0m ignoring /home/conda/feedstock_root/build_artifacts/pydantic-core_1703286521520/work/python/pydantic_core/py.typed: Ignore(IgnoreMatch(Gitignore(Glob { from: Some("/home/conda/feedstock_root/.gitignore"), original: "*", actual: "**/*", is_whitelist: false, is_only_dir: false })))
```

i.e. `maturin` is picking up a `.gitignore` from below the source tree (`/home/conda/feedstock_root/.gitignore`), which has a very aggressive ignore pattern in it, thus causing all contents of the repository to be ignored.

This would explain why it appears to have broken on thin air but also why macOS is still building ok; it must have a different builder layout.

I need to go to sleep now but hopefully this is enough for someone else to figure out what the right fix should be.

**Comment by davidhewitt** [comment]
@hmaarrfk I guess this also explains why downgrading doesn't help.

**Comment by hmaarrfk** [comment]
Oh. Yeah. That was designed by @isuruf maybe he can help.

**Pull request closed without merging**